pkg/systemd/overview-cards/insights.jsx

```diff
@@ -0,0 +1,172 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import cockpit from "cockpit";
+import * as service from "service.js";
+
+import "./insights.scss";
+
+const _ = cockpit.gettext;
+
+const InsightsIcon = ({ critical, important, moderate, low }) => {
+    const vars = {
+        "--critical": critical,
+        "--important": important,
+        "--moderate": moderate,
+        "--low": low
+    };
+    return (
+        <span className="insights-icon" style={vars}>
+            <span className="insights-icon-critical" />
+            <span className="insights-icon-important" />
+            <span className="insights-icon-moderate" />
+            <span className="insights-icon-low" />
+        </span>);
+};
+
+export class InsightsStatus extends React.Component {
+    constructor() {
+        super();
+        this.state = { };
+    }
+
+    componentDidMount() {
+        this.insights_client_timer = service.proxy("insights-client.timer");
+        this.insights_client_timer.addEventListener("changed", () => this.setState({}));
+
+        const watch = (name, state, parser) => {
+            return cockpit.file(name, { syntax: { parse: parser } }).watch((data, tag, error) => {
+                if (error)
+                    console.warn("Parse error", name, error.toString());
+                this.setState({ [state]: data });
+            });
+        };
+
+        this.id_watch = watch("/var/lib/insights/host-details.json", "id",
+                              data => JSON.parse(data).results[0].id);
+
+        this.hits_watch = watch("/var/lib/insights/insights-details.json", "hits",
+                                data => {
+                                    const json = JSON.parse(data);
+                                    const n_hits = json.length;
+                                    const n_hits_by_risk = [0, 0, 0, 0];
+                                    json.forEach(r => {
+                                        const risk_index =
+                                              Math.max(1, Math.min(5, Math.floor(r.rule.total_risk || 0))) - 1;
+                                        n_hits_by_risk[risk_index] += 1;
+                                    });
+                                    return {
+                                        n: n_hits,
+                                        n_by_risk: n_hits_by_risk
+                                    };
+                                });
+    }
+
+    componentWillUnmount() {
+        this.id_watch.remove();
+        this.hits_watch.remove();
+    }
+
+    render() {
+        const subfeats = (cockpit.manifests.subscriptions && cockpit.manifests.subscriptions.features) || { };
```

Comment:
Please move that into the constructor, it doesn't change during the page's lifetime. It only uses the `.insights` flag, so the constructor could just set a `this.register_supported = true` or so.